Sources/StreamVideo/Controllers/CallController.swift

```diff
@@ -75,137 +197,131 @@ class CallController: @unchecked Sendable {
         ring: Bool = false,
         notify: Bool = false
     ) async throws -> JoinCallResponse {
-        let response = try await joinCall(
-            create: create,
-            callType: callType,
-            callId: callId,
-            options: options,
-            migratingFrom: migratingFrom,
-            ring: ring,
-            notify: notify
-        )
-
-        currentSFU = response.credentials.server.edgeName
-        statsInterval = TimeInterval(response.statsOptions.reportingIntervalMs / 1000)
-        let settings = callSettings ?? response.call.settings.toCallSettings
-        
-        try await connectToEdge(
-            response,
-            sessionID: sessionID,
-            callType: callType,
-            callId: callId,
-            callSettings: settings,
-            ring: ring,
-            migratingFrom: migratingFrom
-        )
-        
-        setupStatsTimer()
-        
+        try await webRTCCoordinator.connect(callSettings: callSettings)
+//        try await webRTCClient.connect(callSettings: callSettings)
+        guard let response = try await joinCallResponseSubject.nextValue(dropFirst: 1, timeout: 15) else {
```

Comment:
Because the first value we receive is the currentValue of the subject (on the first access the first value will be nil). We rather want to the next available value that will be received as the connect flow (we started on the live above) proceeds.